OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -143,32 +139,19 @@ public override bool Tick(Actor self)
 			var pos = self.CenterPosition;
 			var checkTarget = useLastVisibleTarget ? lastVisibleTarget : target;
 
-			// We don't know where the target actually is, so move to where we last saw it
-			if (useLastVisibleTarget)
-			{
-				// HACK: Bot players ignore the standard visibility checks in target.Recalculate,
-				// which means that targetIsHiddenActor is always false, allowing lastVisibleMaximumRange
-				// to be assigned zero range by attackAircraft.GetMaximumRangeVersusTarget for e.g. cloaked actors.
-				// Catch and cancel this edge case to avoid the aircraft stopping mid-air!
-				if (self.Owner.IsBot && lastVisibleMaximumRange == WDist.Zero)
```

Comment:
are you sure we aren't regressing #19036 ?

I have run an Europe bot game, and AI aircraft seem to completely ignore submarines. I wonder wether that's regressions or it's already the case in bleed. Some better testing with custom maps would be good. #3468 I think is fixed already?

Reply:
The same check for zero lastVisibleMaximumRange is now run even if the player is not a bot so regressing #19036 seems unlikely.